atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1014,6 +1015,35 @@ public void cleanup() {}
         assertEquals(ImmutableList.of(firstCell, secondCell), cells);
     }
 
+    @Test
+    public void commitDoesNotThrowIfAlreadySuccessfullyCommitted() {
```

Comment:
what should happen if we try to commit, but somebody has rolled us back - is that case covered?